### PR TITLE
SE-1095-Add-licensing-information

### PR DESCRIPTION
**Pull request by ShaneDuncan602**
Also updated foundry version to nightly-5ac78a9cd4b94dc53d1fe5e0f42372b28b5a7559

### CODE REVIEW

File: src/example/LICENSE [hunk 1]
```diff
@@ -0,0 +1,445 @@
...
+
+Additional Use Grant: Any uses listed and defined at
+                      v3-core-license-grants.uniswap.eth
+
+Change Date:          The earlier of 2023-04-01 or a date specified at
+                      v3-core-license-date.uniswap.eth
+
+Change License:       GNU General Public License v2.0 or later
+
+-----------------------------------------------------------------------------
```
Comment (TJ-Everett) [inline src/example/LICENSE]: Is this an example section or should we update this?

**Review by TJ-Everett — COMMENTED** [review]
Just a quick question on that section for the example licensing

File: src/example/LICENSE (hunk 1 above)
Comment (ShaneDuncan602) [reply src/example/LICENSE]: Good catch. This needs to be updated.

**Review by TJ-Everett — APPROVED** [review]
Approved. Great work on this

**Pull request merged**